### PR TITLE
Deduplicating crate dependencies (part 2 of n, `slab`)

### PR DESCRIPTION
**Pull request by shergin**
The change includes only `slab` module.

#11468

### CODE REVIEW

**Comment by shergin** [comment]
cc @vorot93, @ordian, @niklasad1

File: util/network-devp2p/src/host.rs [hunk 1]
```diff
@@ -672,21 +672,18 @@ impl Host {
...
-			match Session::new(io, socket, token, id, &nonce, &self.info.read()) {
-				Ok(s) => Some(Arc::new(Mutex::new(s))),
-				Err(e) => {
-					debug!(target: "network", "Session create error: {:?}", e);
-					None
-				}
-			}
-		});
+		let entry = sessions.vacant_entry();
+		let key = entry.key();
```
Comment (ordian) [inline util/network-devp2p/src/host.rs:676]: is it possible to move these vars under the `Ok(session) =>` match below?

Comment (ordian) [reply util/network-devp2p/src/host.rs:676]: ah, I see it's used in the `trace!` now, it's fine as is then

Comment (shergin) [reply util/network-devp2p/src/host.rs:676]: Thank you for careful review! That are *literately* my first lines of code written in Rust. :)

File: util/network-devp2p/src/host.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ use crate::{
 	session::{Session, SessionData}
 };
 
-type Slab<T> = ::slab::Slab<T, usize>;
+type Slab<T> = ::slab::Slab<T>;
```
Comment (dvdplm) [inline util/network-devp2p/src/host.rs:58]: dq: Why do we need this type alias?

Comment (shergin) [reply util/network-devp2p/src/host.rs]: Oh, we don't need it!
I fixed it.

**Pull request merged**